src/components/UserName.tsx

```diff
@@ -1,38 +1,57 @@
-import {useState} from "react";
+import {useState, useEffect} from "react";
 import "./userName.scss";
+import {useAppDispatch, useAppSelector} from "../store/hooks";
+import {getUser} from "../store/loginName/userSlice";
+import {AnyAction, Dispatch} from "redux";
+import classNames from 'classnames';
 
 type UserNameProps = {
   minSymbol: number;
   maxSymbol: number;
 }
-const UserName = ({minSymbol, maxSymbol}: UserNameProps):JSX.Element => {
-  const userNameRegex = new RegExp('^[A-Z\d]{' + minSymbol + ',' + maxSymbol + '}$', 'i');
+const UserName = ({minSymbol, maxSymbol}: UserNameProps): JSX.Element => {
+  const userNameRegex = new RegExp('^[A-ZА-ЯЁ \d]{' + minSymbol + ',' + maxSymbol + '}$', 'i');
   const [value, setValue] = useState('');
-  const [isError, setIsError] = useState(false);
   const [isBlur, setIsBlur] = useState(false);
+  const [extraStyle, setExtraStyle] = useState('');
+  const [error, setError] = useState('');
+
+  const dispatch = useAppDispatch();
+  const userStatus = useAppSelector((state) => state.user.isLogged);
 
   const fieldHandler = (e: React.FormEvent<HTMLInputElement>) => {
-    if (userNameRegex.test(e.currentTarget.value)) {
-      setIsError(false);
+    const value = e.currentTarget.value;
+    setValue(value);
+    dispatch(getUser(value));
+
+    if (!userNameRegex.test(value)) {
+      setError('Invalid username entered');
+      setExtraStyle('errorInput');
     } else {
-      setIsError(true);
+      setError('');
     }
-    setValue(e.currentTarget.value);
   }
 
+  useEffect(() => {
+    if (userStatus) {
+      setError('User already exists. Please enter a different username');
+      setExtraStyle('redBorder')
+    }
+  }, [userStatus]);
+
   const blurHandle = (e: React.FormEvent<HTMLInputElement>) => {
```

Comment:
const blurHandle = (e: React.FormEvent<HTMLInputElement>) => {
 
 Remove e from arguments. you do not use it